### PR TITLE
[MM-16376] Add installPluginFromUrl to Client and Actions

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

This PR adds a new API method to install a plugin through providing a download URL to the server.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-16376

#### Checklist

- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit tests passed
- [x] Ran `make flow` to ensure type checking passed
- [x] Added or updated unit tests (required for all new features)
- [x] All new/modified APIs include changes to the [JavaScript driver](https://github.com/mattermost/mattermost-redux/blob/master/src/client/client4.js)

#### Test Information

Tested on Chrome in Ubuntu.

#### Dependency

https://github.com/mattermost/mattermost-server/pull/11372

### CODE REVIEW

File: src/actions/admin.js [hunk 1]
```diff
@@ -461,6 +461,30 @@ export function uploadPlugin(fileData: File, force: boolean = false): ActionFunc
...
+export function installPluginFromUrl(url: string, force: boolean = false): ActionFunc {
+    return async (dispatch, getState) => {
+        dispatch({type: AdminTypes.INSTALL_PLUGIN_FROM_URL_REQUEST, data: null});
+
+        let data;
+        try {
+            data = await Client4.installPluginFromUrl(url, force);
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch, getState);
+            dispatch(batchActions([
```
Comment (cpoile) [inline src/actions/admin.js]: Do we need to send this as a batch action (if it's just one action)?

File: src/actions/admin.js [hunk 2]
```diff
@@ -461,6 +461,30 @@ export function uploadPlugin(fileData: File, force: boolean = false): ActionFunc
...
+                {type: AdminTypes.INSTALL_PLUGIN_FROM_URL_FAILURE, error},
+                logError(error),
+            ]));
+            return {error};
+        }
+
+        dispatch(batchActions([
```
Comment (cpoile) [inline src/actions/admin.js]: Same question

File: src/actions/admin.js (hunk 1 above)
Comment (mickmister) [reply src/actions/admin.js]: We sure don't :wink:

Good catch!

Comment (mickmister) [reply src/actions/admin.js]: @cpoile Actually this one is dispatching two actions, but the one your comment below references should be fixed.

Comment (mickmister) [reply src/actions/admin.js]: Also, it seems other actions in this file are doing the same unnecessary `batchActions()` call, not that it matters for the scope of this PR.

**Comment by cpoile** [comment]
@mickmister waiting for tests

**Pull request merged**